src/main/scala/subsystem/HasTiles.scala

```diff
@@ -10,25 +10,47 @@ import freechips.rocketchip.devices.tilelink.{BasicBusBlocker, BasicBusBlockerPa
 import freechips.rocketchip.diplomacy._
 import freechips.rocketchip.diplomaticobjectmodel.logicaltree.{LogicalModuleTree}
 import freechips.rocketchip.interrupts._
-import freechips.rocketchip.tile.{BaseTile, LookupByHartIdImpl, TileParams, HasExternallyDrivenTileConstants, InstantiableTileParams, PriorityMuxHartIdFromSeq}
+import freechips.rocketchip.tile.{BaseTile, LookupByHartIdImpl, TileParams, InstantiableTileParams, MaxHartIdBits}
 import freechips.rocketchip.tilelink._
 import freechips.rocketchip.util._
 
 /** Entry point for Config-uring the presence of Tiles */
 case class TilesLocated(loc: HierarchicalLocation) extends Field[Seq[CanAttachTile]](Nil)
 
+/** Whether a global hart id prefix is going to be dynamically applied to the static tile hart ids.
+  *
+  * This information is used to conditionally add timing-closure registers along the path of
+  * the dynamically-configurable hart id as it makes its way into the tile.
```

Comment:
```suggestion
  * the dynamically-configurable hart id prefix as it makes its way into the tile.
```

Reply:
Looking at how this is used, shouldn't this be called something like "RegisterHartPrefixKey" ? It has nothing to do with whether to actually apply the prefix or not, that's given by whoever connects to tileHartIdeNexusNode

Reply:
Previously it did both, now it only does the what comment says. I just kept the name the same since closed source is using it, but you're right that it would be better to update it and it's usages to be more clearly named. I'll change the name but I don't agree with your comment; most of the registers that are added contain the entire bit-wise-or'd hart id, not just the prefix.

Reply:
cool